fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/PartitionAvailability.java

```diff
@@ -0,0 +1,158 @@
+ * | [2017-01/2017-02]   |  [2017/2018]         |
+ * +---------------------+----------------------+
+ *
+ * If constraints participates in all parts:
+ *
+ * +-------------------------+
+ * |  availability(1,2)      |
+ * +-------------------------+
+ * | [2017-01/2017-02]       |
+ * +-------------------------+
+
+ * If a query participates only in the larger partition
+ *
+ * +-------------------------+
+ * |  availability(2)        |
+ * +-------------------------+
+ * | [2017-01/2018]          |
+ * +-------------------------+
+ * }
+ * </pre>
+ */
+public class PartitionAvailability extends BaseCompositeAvailability implements Availability {
+
+    private final Map<Availability, DataSourceFilter> availabilityFilters;
+    private final Set<TableName> dataSourceNames;
+    /**
+     * Constructor.
+     *
+     * @param availabilityFilters  Filters to identify which availabilities apply for a set of constraints
+     * Availabilities. The function is composed of the following works functions/transformations
+     * <pre>
+     * {@code
+     * 1. Function(DataSourceConstraint) -> Boolean
+     *
+     *
+     * 2. Function(ApiFilter) -> sef of PartitionKeys:
+     *        Dimension
+     *        Map<DimensionRow, PartitionKey>
+     *
+     *        SingleDimensionValueMap(Dimension, Map<DimensionRow, PartitionKey>)
+     *
+     *        apply():
+     *            dimension.getSearchProvider(ApiFilters)
+     *                    .findFilteredDimensionRowsPaged(
+     *                            apiFilters,
+     *                            PaginationParameters.EVERYTHING_IN_ONE_PAGE
+     *                    ).stream()
+     *                    .map(map::get)  // This will fail in potentially bad ways if the key isn't mapped
+     *                    .collect(Collectors.toSet());
+     *
+     * 3. Map<PartitionKey, Availability>
+     * }
+     * </pre>
+     */
+    public PartitionAvailability(
+            @NotNull Map<Availability, DataSourceFilter> availabilityFilters
+    ) {
+        this.availabilityFilters = availabilityFilters;
+        this.dataSourceNames = super.getDataSourceNames();
+    }
+
+    @Override
+    protected Stream<Availability> getAllDependentAvailabilities() {
+        return availabilityFilters.keySet().stream();
+    }
+
+    @Override
+    public Set<TableName> getDataSourceNames() {
+        return dataSourceNames;
+    }
+
+    /**
+     * Retrieve all available intervals for all child availabilities.
+     * <p>
+     * Available intervals for the same datasource columns are unioned into a <tt>SimplifiedIntervalList</tt>
+     *
+     * @return a map of column to all of its available intervals in union
+     */
+    @Override
+    public Map<String, SimplifiedIntervalList> getAllAvailableIntervals() {
+        // get all availabilities take available interval maps from all availabilities and merge the maps together
+        return availabilityFilters.keySet().stream()
```

Comment:
Other than getting this from the keyset rather than a different list, this sure _looks_ like it's exactly the same implementation as the parent class. Can we dedupe?